### PR TITLE
Allow static build of lib-express to simplify application deployment

### PR DESCRIPTION
**Pull request by daminetreg**
Hey @linkineo and how is the weekend ? :smile:

I just made some change to the build files for the purpose of deploying more easily application built on top of express : 
  - allow static build.
  - avoid compilation of unit test.

See you soon

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
+
+option (UNIT_TESTS "Enable Unit Tests" OFF)
 
 set (VERSION_MAJOR 0)
 set (VERSION_MINOR 1)
 set (VERSION_PATCH 0)
 
 set (VERSION_STRING_INSTALL "${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}")
 
-add_definitions(--std=c++11)
```
Comment (linkineo) [inline CMakeLists.txt]: Why do you remove the C++11 flag ? When doing so, it still compiles on travis as I'm using new compilers with this setting as default, but on OS X this is not default yet and fails, so I need it.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
+if (NOT BUILD_SHARED_LIBRARIES) 
+  SET(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
+endif() 
 
-find_package(Boost 1.59 REQUIRED COMPONENTS filesystem regex system thread filesystem date_time)
+find_package(Boost 1.59 REQUIRED COMPONENTS regex coroutine context thread filesystem system date_time)
```
Comment (linkineo) [inline CMakeLists.txt]: I just tested again, SimpleWebServer as integrated in express no longer needs coroutine or context. It compiles without them for sure: 
`c++   -dynamiclib -Wl,-headerpad_max_install_names -current_version 0.1.0 -o libexpress-cpp.0.1.0.dylib -install_name /Users/linkineo/Devl/tests/express-cpp/build/libexpress-cpp.0.1.0.dylib CMakeFiles/express-cpp.dir/src/application/application.cpp.o CMakeFiles/express-cpp.dir/src/request/request.cpp.o CMakeFiles/express-cpp.dir/src/response/response.cpp.o /usr/local/lib/libboost_regex-mt.a /usr/local/lib/libboost_thread-mt.a /usr/local/lib/libboost_filesystem-mt.a /usr/local/lib/libboost_system-mt.a /usr/local/lib/libboost_date_time-mt.a`

File: CMakeLists.txt [hunk 3]
```diff
@@ -23,13 +25,16 @@ include_directories(${SIMPLE_WEB_SERVER_HDR})
 
 
 file(GLOB_RECURSE EXPRESS_SRC "src/*.cpp")
-add_library(express-cpp SHARED ${EXPRESS_SRC})
```
Comment (linkineo) [inline CMakeLists.txt:26]: Well, I'm ok for static build support as an OPTION, but not as a default, which you instated by doing this. It's ok for the user to say something like -DBUILD_SHARED_LIBS=OFF, but when not specifying any option, it has to be a shared library.

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
 
 set (VERSION_MAJOR 0)
 set (VERSION_MINOR 1)
 set (VERSION_PATCH 0)
 
 set (VERSION_STRING_INSTALL "${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}")
 
-add_definitions(--std=c++11)
+if (NOT BUILD_SHARED_LIBRARIES) 
+  SET(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
```
Comment (linkineo) [inline CMakeLists.txt]: It this portable ?

**Review by linkineo — CHANGES_REQUESTED** [review]
Hello Damien,

Thanks for your contribution. I added a few comments and change requests.

See you
Patrick

File: CMakeLists.txt (hunk 4 above)
Comment (daminetreg) [reply CMakeLists.txt]: Oh yes I forgot to add if (linux). Good catch.

File: CMakeLists.txt (hunk 2 above)
Comment (daminetreg) [reply CMakeLists.txt]: ok good then we can remove them.

File: CMakeLists.txt (hunk 1 above)
Comment (daminetreg) [reply CMakeLists.txt]: Yes I wanted to replace with CXX_STANDARD.

File: CMakeLists.txt (hunk 3 above)
Comment (daminetreg) [reply CMakeLists.txt:26]: Why not static by default, it's actually the praised way by CMake and massive projects like hunter. It in fact makes application deployment way easier I think.

But I add it per default ON in your project, even though I think it should be OFF per default.

**Comment by linkineo** [comment]
Great, thanks !

**Comment by daminetreg** [comment]
Thanks for merging 😊

--
Damien Buhl
Software Developer

**Pull request merged**